test/membrane/life-red-proxies.spec.js

```diff
@@ -0,0 +1,67 @@
+import createSecureEnvironment from '../../lib/browser-realm.js';
+
+const LockerLiveValueMarkerSymbol = Symbol.for('@@lockerLiveValue');
+
+const o = {
+    x: 'uno'
+};
+Object.defineProperty(o, LockerLiveValueMarkerSymbol, {});
+
+const endowments = {
+    o,
+    expect,
+};
+const evalScript = createSecureEnvironment(undefined, endowments);
+
+describe('A Live Red Proxy', () => {
+    it('should surface new expandos from blue realm', function() {
+        // expect.assertions(2);
+        o.x = 'uno';
+        o.y = 'dos';
+        evalScript(`
+            expect(o.x).toBe('uno');
+            expect(o.y).toBe('dos');
+        `);
+    });
+    it('should allow mutation from blue realm', function() {
+        // expect.assertions(4);
+        o.x = 'tres';
+        o.y = 'cuatro';
+        evalScript(`
+            expect(o.x).toBe('tres');
+            expect(o.y).toBe('cuatro');
+        `);
+        expect(o.x).toBe('tres');
+        expect(o.y).toBe('cuatro');
+    });
+    it('should allow mutation from within the sandbox', function() {
+        // expect.assertions(4);
+        evalScript(`
+            o.x = 'cinco';
+            o.y = 'six';
+            expect(o.x).toBe('cinco');
+            expect(o.y).toBe('six');
+        `);
+        expect(o.x).toBe('cinco');
+        expect(o.y).toBe('six');
+    });
+    it('should allow expandos added form within the sandbox', function() {
+        // expect.assertions(2);
+        evalScript(`
+            o.z = 'seven';
+            expect(o.z).toBe('seven');
+        `);
+        expect(o.z).toBe('seven');
+    });
+    it('should only have effect on own properties', function() {
```

Comment:
👍

Reply:
I'm a little puzzled about 2 scenarios, maybe this is by design but I think they are worth looking into.

the following code seems to work but by looking at this test it shouldn't

```
        o.w = {};
        evalScript(`
            o.w = {a:1}
        `);
        expect(o.w.a).toBe(1); // based on the test {a:1} should be an expando right ?
```

The other thing I found interesting was being able to get the symbol and attach it to other objects and make them reactive from within the sandbox. Observe the following code:

```
        const obj = {};
        Object.defineProperty(obj, Symbol.for('@@lockerLiveValue'), {});
        const env1 = createSecureEnvironment(undefined, {obj});
        env1(`
            const symbol = Object.getOwnPropertySymbols(obj);
            obj.foo = {};
            Object.defineProperty(obj.foo, symbol[0], {});
        `);

        const env2 = createSecureEnvironment(undefined, {foo: obj.foo}); // notice I am sending obj.foo, an object which the blue environment knows it shouldn't be dynamic.
        env2('foo.x = "bar"');
        console.log(obj.foo) // {x: 'bar'}
```

Reply:
the first scenario I didn't get it, it seems that the code is incomplete.

the second, yes, that also works fine... u don't need to get the symbol out of the obj, you can use Symbol.for, since they are shared between realms.

that should not pose a problem at all... in fact, red to blue is already dynamic... when u get it back to you, you get the original, and if blue changes it, u will see it even without the symbol, the symbol only adds another dimension, if that obj.foo is given to another namespace, it is considered live :)